JPATask/src/main/java/com/instinctools/egor/mentoring/JPA/DAO/DAOImpl/ClientDAOImpl.java

```diff
@@ -0,0 +1,95 @@
+package com.instinctools.egor.mentoring.JPA.DAO.DAOImpl;
+
+import com.instinctools.egor.mentoring.JPA.DAO.ClientDAO;
+import com.instinctools.egor.mentoring.JPA.DAO.DAOUtil.DAOUtil;
+import com.instinctools.egor.mentoring.JPA.Entity.Account;
+import com.instinctools.egor.mentoring.JPA.Entity.Client;
+
+import javax.persistence.EntityManager;
+import java.sql.SQLException;
+import java.util.List;
+
+public class ClientDAOImpl implements ClientDAO {
+    public void createClient(Client client) throws SQLException {
+        EntityManager session = DAOUtil.getSessionFactory().createEntityManager();
+        try {
+            session.getTransaction().begin();
+            session.persist(client);
+            session.getTransaction().commit();
+        }
+        finally {
+            if(session != null && session.isOpen()){
+                session.close();
+            }
+        }
+    }
+
+    public void deleteClient(Client client) throws SQLException {
+        EntityManager session = DAOUtil.getSessionFactory().createEntityManager();
+        try {
+            session.getTransaction().begin();
+            session.remove(session.contains(client) ? client : session.merge(client));
+            session.flush();
+            session.getTransaction().commit();
+        }
+        finally {
+            if(session != null && session.isOpen()){
+                session.close();
+            }
+        }
+    }
+
+    public List<Client> getAllClients() throws SQLException {
+        EntityManager session = DAOUtil.getSessionFactory().createEntityManager();
+        List<Client> clients = null;
+        try {
+            session.getTransaction().begin();
+            clients = session.createQuery("from Client", Client.class).getResultList();
+            session.getTransaction().commit();
+        }
+        finally {
+            session.close();
+        }
+        return clients;
+    }
+
+    public void createAccount(Client client, Account account) throws SQLException {
+        EntityManager entityManager = DAOUtil.getSessionFactory().createEntityManager();
+        try {
+           entityManager.getTransaction().begin();
+           client.addAccount(account);
+           entityManager.merge(client);
+           entityManager.flush();
```

Comment:
the same about flush()
and we don't need this at all - please check my comment on `ClientDAO` interface